tests/foreman/sys/test_katello_certs_check.py

```diff
@@ -332,3 +336,67 @@ def test_positive_validate_without_req_file_output(self):
 
         :CaseAutomation: notautomated
         """
+
+    @tier1
+    def test_positive_validate_capsule_certificate(self):
+        """Check that Capsules cert handles additional proxy names.
+
+        :id: 8b53fc3d-704f-44f4-899e-74654529bfcf
+
+        :steps:
+
+            1. Generate a Capsule certificate
+            2. Confirm proxy server's FQDN for DNS is present
+            3. Confirm that format of alternative names does not include []
+
+        :expectedresults: Capsule certs has valid DNS values
+
+        :BZ: 1747581
+
+        :CaseAutomation: automated
+        """
+        DNS_Check = False
+        tmp_dir = '/var/tmp/{0}'.format(gen_string('alpha', 4))
+        cert_file = '{0}/ssl-build/capsule.example.com/cert-data'.format(tmp_dir)
+        if not os.path.exists('{0}/ssl-build/capsule.example.com/'.format(tmp_dir)):
+            os.makedirs('{0}/ssl-build/capsule.example.com/'.format(tmp_dir))
+        with get_connection(timeout=200) as connection:
+            result = connection.run(
+                'mkdir {0}'.format(tmp_dir))
+            assert result.return_code == 0, 'Create working directory failed.'
+            result = connection.run(
```

Comment:
It would be more cleaner if we do setup part separate from test case. I would suggest to introduce separate class and same setup you can use for adding other tests related to capsule-certs-generate. ex: https://bugzilla.redhat.com/show_bug.cgi?id=1709761

Reply:
Hello @ntkathole , I am trying to see the benefit of moving  some lines to a new class. The only things I can see to move out to a class setup are `tmp_dir` and `cert_file` setup. But there is no Capsule test, even a stubbed one, that would use that.  The part you highlight is creating a tmp dir on Satellite base system for this test. It would not be used in a full installation test for Capsule certs (as its on the Satellite).

Looking at BZ#1709761, I wonder do you mean we could add a test for that later? 

Anyway, I have made a new class and left it as a separate git commit for ease of review. Please review.

Thank you

Reply:
@swadeley yes, I mean if going forward we have added coverage for BZ#1709761 then the setup  class should be reusable.

In this case, can we move capsule-certs-generate command to setup class if you also see benefits of doing so? else I am ok with current changes you did.

Reply:
OK, done. Thank you.